### PR TITLE
Inheritance

### PR DESCRIPTION
**Pull request by KarthikMudhigonda**
Requesting for code review.

### CODE REVIEW

File: src/inheritance/Address.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package inheritance;
+
+public class Address {
+    String city,state,country;
+    
+public Address(String city, String state, String country){
+        this.city = city;
+        this.state = state;
+        this.country = country;
+
```
Comment (shylendramadda) [inline src/inheritance/Address.java:10]: Remove this extra space

File: src/inheritance/Emp.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    int id;
+    String name;
+    Address address;
+
+public Emp(int id, String name, Address address){
+    this.id = id;
+    this.name = name;
+    this.address = address;
+}
+void display(){
```
Comment (shylendramadda) [inline src/inheritance/Emp.java:13]: Giva space before starting of a new method and after completion of the method

File: src/inheritance/TestInheritance.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package inheritance;
+
+class Animal{
+    void eat(){System.out.println("eating...");}
+}
+class Dog extends Animal{
```
Comment (shylendramadda) [inline src/inheritance/TestInheritance.java:6]: Give a space between 2 classes

File: src/inheritance/TestInheritance2.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+public class TestInheritance2 {
+    public static void main(String[] args) {
+        BabyDog b = new BabyDog();
+        b.bark();
+        b.eat();
+        b.weep();
+
+
```
Comment (shylendramadda) [inline src/inheritance/TestInheritance2.java:27]: Remove extra empty lines, empty line also consumes some memory

File: src/inheritance/Address.java (hunk 1 above)
Comment (KarthikMudhigonda) [reply src/inheritance/Address.java:10]: Resolved the issue

File: src/inheritance/TestInheritance.java (hunk 3 above)
Comment (KarthikMudhigonda) [reply src/inheritance/TestInheritance.java]: Resolved the issue

File: src/inheritance/TestInheritance2.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        System.out.println("eating...");
+    }
+}
+
+class Dog extends Animal {
+    void bark() {
+        System.out.println("barking..");
+    }
+}
+class BabyDog extends Dog{
```
[the remark this answers is not in the document]
Comment (KarthikMudhigonda) [reply src/inheritance/TestInheritance2.java]: Resolved the issue

**Review by shylendramadda — APPROVED** [review]

**Pull request merged**